### PR TITLE
Set admin for config, in restaking and vault

### PR DESCRIPTION
**Pull request by ebatsell**

### CODE REVIEW

File: vault_sdk/src/instruction.rs [hunk 1]
```diff
@@ -145,7 +145,7 @@ pub enum VaultInstruction {
     },
 
     #[account(0, name = "config")]
-    #[account(1, name = "vault")]
+    #[account(1, writable, name = "vault")]
```
Comment (buffalu) [inline vault_sdk/src/instruction.rs:148]: accident?

File: vault_program/src/set_config_admin.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+    Config::load(program_id, config, true)?;
+    let mut config_data = config.data.borrow_mut();
+    let config = Config::try_from_slice_unchecked_mut(&mut config_data)?;
+    load_signer(old_admin, false)?;
+    load_signer(new_admin, false)?;
+
+    if config.admin != *old_admin.key {
+        return Err(VaultError::VaultConfigAdminInvalid.into());
```
Comment (buffalu) [inline vault_program/src/set_config_admin.rs:22]: config admin invalid (not a vault)

File: vault_sdk/src/instruction.rs (hunk 1 above)
Comment (ebatsell) [reply vault_sdk/src/instruction.rs:148]: good catch left over from another PR

File: vault_program/src/set_config_admin.rs (hunk 2 above)
Comment (ebatsell) [reply vault_program/src/set_config_admin.rs:22]: renamed to just ConfigAdminInvalid

**Review by coachchucksol — APPROVED** [review]
lg2m

File: rust-toolchain.toml [hunk 3]
```diff
@@ -1,4 +1,3 @@
 # note: this file doesn't play nicely with solana-verify build
 [toolchain]
-channel = "1.80.0"
```
Comment (buffalu) [inline rust-toolchain.toml:3]: nit

**Review by buffalu — APPROVED** [review]

**Pull request merged**